### PR TITLE
Make more APIs jsonapi compatible

### PR DESCRIPTION
**Pull request by sehraf**
Including small  cleanup

### CODE REVIEW

**Comment by sehraf** [comment]
This PR makes my always beloved Lua script obsolete :cry:
[buryTheDead.lua](https://github.com/RetroShare/Lua4RS/blob/master/scripts/buryTheDead.lua) is dead, long live [buryTheDead.py](https://gist.github.com/sehraf/4774b0f45311a2154d1f7f648ddbed6e) 
@HLT

**Comment by sehraf** [comment]
There goes another Lua script ...
[friendsVersionOverview.lua](https://github.com/RetroShare/Lua4RS/blob/master/scripts/friendsVersionOverview.lua) :arrow_forward: [discOverview.py](https://gist.github.com/sehraf/7bce1eccbb1eee8fa277560522eb2ca1)

File: libretroshare/src/retroshare/rsconfig.h [hunk 1]
```diff
@@ -129,9 +134,8 @@ int 		promptAtBoot; /* popup the password prompt */
 };
 
 
-class RsConfigDataRates
+struct RsConfigDataRates : RsSerializable
 {
-	public:
 	RsConfigDataRates()
 	{
 		mRateIn = 0;
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsconfig.h]: This should use proper initializer : mRateIn(0) if initialization is needed

File: libretroshare/src/retroshare/rsconfig.h [hunk 2]
```diff
@@ -180,11 +202,22 @@ class RSTrafficClue
 
     RSTrafficClue() { TS=0;size=0;service_id=0;service_sub_id=0; count=0; }
     RSTrafficClue& operator+=(const RSTrafficClue& tc) { size += tc.size; count += tc.count ; return *this ;}
+
+	// RsSerializable interface
+public:
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsconfig.h]: public can be removed here as it is a struct and averything is public by default

File: libretroshare/src/retroshare/rsconfig.h [hunk 3]
```diff
@@ -195,33 +228,63 @@ class RsConfigNetStatus
...
+	bool		netUpnpOk;		/* upnp is enabled and active */
+	bool		netDhtOk;		/* response from dht */
+	bool		netStunOk;		/* recvd stun / udp packets */
+	bool		netExtAddressOk;/* from Dht/Stun or External IP Finder */
+
+	uint32_t	netDhtNetSize;	/* response from dht */
+	uint32_t	netDhtRsNetSize;/* response from dht */
+
+	// RsSerializable interface
+public:
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsconfig.h]: public can be removed here as it is a struct and averything is public by default

File: libretroshare/src/retroshare/rsdisc.h [hunk 4]
```diff
@@ -30,20 +30,55 @@
...
+ */
 extern RsDisc   *rsDisc;
 
 class RsDisc
 {
 	public:
 
 	RsDisc()  { return; }
-virtual ~RsDisc() { return; }
+	virtual ~RsDisc() { return; }
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsdisc.h]: now that you edited the line please take the chance to remove the horrible `return;` on the empty method

File: libretroshare/src/retroshare/rsservicecontrol.h [hunk 5]
```diff
@@ -54,32 +60,42 @@ class RsServiceInfo
 	uint16_t    mMinVersionMajor;
 	uint16_t    mMinVersionMinor;
 
-	static unsigned int RsServiceInfoUIn16ToFullServiceId(uint16_t serviceType);
+	// RsSerializable interface
+public:
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsservicecontrol.h]: Remove the unneeded `public:` it is a struct so everything is public by default

File: libretroshare/src/retroshare/rsservicecontrol.h [hunk 6]
```diff
@@ -54,32 +60,42 @@ class RsServiceInfo
...
-	RsPeerServiceInfo()
-	:mPeerId(), mServiceList() { return; }
+	RsPeerServiceInfo() : mPeerId(), mServiceList() {}
 
-        RsPeerId mPeerId;
+	RsPeerId mPeerId;
 	std::map<uint32_t, RsServiceInfo> mServiceList;
+
+	// RsSerializable interface
+public:
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsservicecontrol.h]: Remove the unneeded `public:` it is a struct so everything is public by default

File: libretroshare/src/retroshare/rsservicecontrol.h [hunk 7]
```diff
@@ -96,26 +112,95 @@ class RsServicePermissions
     // if DefaultAllowed = false, then only PeersAllowed is checked.
     std::set<RsPeerId> mPeersAllowed;
     std::set<RsPeerId> mPeersDenied;
+
+	// RsSerializable interface
+public:
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsservicecontrol.h]: Remove the unneeded `public:` it is a struct so everything is public by default

File: libretroshare/src/retroshare/rsservicecontrol.h [hunk 8]
```diff
@@ -96,26 +112,95 @@ class RsServicePermissions
...
-virtual bool getServiceItemNames(uint32_t service_id,std::map<uint8_t,std::string>& names) = 0;
-
-virtual bool getServicesAllowed(const RsPeerId &peerId, RsPeerServiceInfo &info) = 0;
-virtual bool getServicesProvided(const RsPeerId &peerId, RsPeerServiceInfo &info) = 0;
-virtual bool getServicePermissions(uint32_t serviceId, RsServicePermissions &permissions) = 0;
-virtual bool updateServicePermissions(uint32_t serviceId, const RsServicePermissions &permissions) = 0;
-
-virtual void getPeersConnected(const uint32_t serviceId, std::set<RsPeerId> &peerSet) = 0;
-
+	virtual ~RsServiceControl() { return; }
```
Comment (G10h4ck) [inline libretroshare/src/retroshare/rsservicecontrol.h]: now that you edited the line please take the chance to remove the horrible `return;` on the empty method

File: libretroshare/src/services/p3discovery2.cc [hunk 9]
```diff
@@ -1174,7 +1174,16 @@ bool p3discovery2::getDiscFriends(const RsPeerId& id, std::list<RsPeerId> &proxy
 		}
 	}
 	return true;
-		
+
+}
+
+bool p3discovery2::getWaitingDiscCount(size_t &sendCount, size_t &recvCount)
+{
+	RsStackMutex stack(mDiscMtx); /********** STACK LOCKED MTX ******/
```
Comment (G10h4ck) [inline libretroshare/src/services/p3discovery2.cc]: Please use `RS_STACK_MUTEX(mDiscMtx);` without the obvious comment ;)

**Review by G10h4ck — CHANGES_REQUESTED** [review]
Great work, just few style observation ;)

**Comment by sehraf** [comment]
done. FYI the `public:` statement is generated by Qt Creator and was just too lazy to remove it all the time :wink:

**Comment by G10h4ck** [comment]
Thanks for your contribution!

**Pull request merged**